target/linux/ath79/dts/qca9558_buffalo_wzr-450hp2.dts

```diff
@@ -127,39 +127,193 @@
 &mdio0 {
 	status = "okay";
 
-	phy0: ethernet-phy@0 {
+	switch@0 {
+		compatible = "qca,qca8337";
+		#address-cells = <1>;
+		#size-cells = <0>;
+
 		reg = <0>;
+		qca,ignore-power-on-sel;
+		qca,led-open-drain;
+
+		ports {
+			#address-cells = <1>;
+			#size-cells = <0>;
+
+			port@0 {
+				reg = <0>;
+				ethernet = <&eth1>;
+				phy-mode = "sgmii";
```

Comment:
should be swapped with port 6.

```
git grep -B 6 phy-mode\ =\ \"sgmii | grep port\@0
```

returns no results.

```
git grep -B 6 phy-mode\ =\ \"sgmii | grep port\@6
```

returns plenty.

Reply:
After swapping port 0 and 6, eth0 and wan port work.